### PR TITLE
generics blog post

### PR DESCRIPTION
**Pull request by topepo**

### CODE REVIEW

**Comment by batpigandme** [comment]
I made a little [dummy version](https://github.com/batpigandme/tidyverse.org/commit/300ee363a5a8a4bff6cb2e152e64c86b7a8d6282) (without the copy edits or anything) of what the setup would look like for this post using `.Rmarkdown`. (I found the process to be confusing the first couple times around, so thought it might be helpful. 🙂)

Feel free to pull the grab the images from there if you want, they're sized and compressed.

File: content/articles/2018-11-generics.markdown [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+author: Max Kuhn, Davis Vaughan
+categories: [package]
+description: >
+    General S3 methods
+photo:
+  url: https://unsplash.com/photos/9GMO0Sxyw_Y
+  author: Jonathan Knepper
+---
+
+The `generics` package is now on CRAN . `generics` is a lightweight package that is designed to help developers reduce dependencies for their packages.
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply content/articles/2018-11-generics.markdown]: ✔️

File: content/articles/2018-11-generics.markdown [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+The idea is to have many generic S3 functions live in a single package with almost no dependencies. For example, if you wanted to use a `broom::tidy()` method, you would use `broom` as an import dependency, import the `tidy()` generic, then export it so that your package can have a specific method. One issue with this is that your package now carries along the dependencies of the package that defines the generic (in this case, `broom`).
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply content/articles/2018-11-generics.markdown]: ✔️ 
For example -> Normally

File: content/articles/2018-11-generics.markdown [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Consider `fit()`. There are at least 15 packages that export an object called `fit` (either a function or method) and at least 12 have specific S3 `fit()` methods. When more than one are loaded, there is a potential for conflicts to arise:
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply content/articles/2018-11-generics.markdown]: ✔️

File: content/articles/2018-11-generics.markdown [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+     ‘recipes’
+
+        • ‘check_cols’: ‘check_cols’
+
+        • ‘check_missing’: ‘check_missing’
+
+        • ‘check_range’: ‘check_range’
+
+<snip>
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply content/articles/2018-11-generics.markdown]: ✔️ snipped the dangler

**Pull request merged**